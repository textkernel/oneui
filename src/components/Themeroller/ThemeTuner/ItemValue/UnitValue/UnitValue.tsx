import * as React from 'react';
import { bem } from '../../../../../utils';
import { Input } from '../../../../Input';
import { Text } from '../../../../Text';
import { ThemeUnitItem } from '../../../../../themes/themerollerConfig';
import styles from './UnitValue.scss';

interface Props {
    item: ThemeUnitItem;
    onChange: (value: ThemeUnitItem) => void;
}

const { elem } = bem('UnitValue', styles);

export const UnitValue: React.FC<Props> = ({ item, onChange }) => {
    const handleOnChange = (event) => {
        onChange({
            ...item,
            value: event.target.value,
        });
    };
    return (
        <span>
            <Input
                {...elem('input')}
                type="number"
                size="small"
                value={item.value}
                onChange={handleOnChange}
            />
            <Text {...elem('value')} inline context="muted">
                {item.value}
                {item.unit}
            </Text>
        </span>
    );
};

UnitValue.displayName = 'UnitValue';

UnitValue.defaultProps = {};
