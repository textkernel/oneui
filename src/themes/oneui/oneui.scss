@import './base-variables/colors';
@import './base-variables/typography';
@import './base-variables/sizing';
@import './base-variables/misc';
@import './base-variables/header_footer';

@import './mixins/colors';
@import './placeholders/input';
