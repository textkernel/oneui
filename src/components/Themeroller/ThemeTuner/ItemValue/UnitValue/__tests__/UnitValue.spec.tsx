import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeUnitItem } from '@textkernel/oneui';
import { UnitValue } from '../UnitValue';

describe('UnitValue component', () => {
    const onChangeMock = jest.fn();
    const item = {
        value: '12',
        unit: 'px',
    } as ThemeUnitItem;

    it('should render component correctly', () => {
        const view = render(<UnitValue item={item} onChange={onChangeMock} />);

        // expect(wrapper.find('.UnitValue__value').at(0).text()).toEqual('12px');
        expect(view.container).toMatchSnapshot();
    });

    it.skip('should invoke onChange callback when input is changed', () => {
        render(<UnitValue item={item} onChange={onChangeMock} />);
        // wrapper.find('input').simulate('change', {
        //     target: {
        //         value: '18',
        //     },
        // });
        expect(onChangeMock).toHaveBeenCalledWith({
            value: '18',
            unit: 'px',
        });
    });
});
