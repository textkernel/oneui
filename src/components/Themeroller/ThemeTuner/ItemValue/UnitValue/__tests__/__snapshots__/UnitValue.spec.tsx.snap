// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitValue component should render component correctly 1`] = `
<div>
  <span>
    <input
      class="Input Input--size_small UnitValue__input"
      data-lpignore="true"
      type="number"
      value="12"
    />
    <span
      class="Text--context_neutral UnitValue__value"
    >
      12
      px
    </span>
  </span>
</div>
`;
